wiring/wiring_spi.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wiring
+
+import "github.com/ligato/cn-infra/core"
+
+// Wiring is a function that, when called on a core.Plugin is expected to 'wire'
+// Its dependencies
+type Wiring func (plugin core.Plugin) error
+
+// Wireable implements a Wire function to wire in dependencies using a Wiring
+type Wireable interface {
+	Wire(wiring Wiring) error  // nil Wiring should result in a Default Wiring
+	IsWired() bool // true if the plugin is Wired, false otherwise
```

Comment:
The `IsWired` seems kind of unnecessary, shouldn't `DefaultWiring` take care of the proper initialization if not set otherwise?